### ISSUE TITLE
crash on ios-devices, in simulator all seems to be fine

### ISSUE THREAD

**Issue by flandy84** [CC-Archived/CCMultipartReader#1]
Hi,

thanks again for that great wrapper.
Aself-explanatory Obj-C API as well.

I 've got a small problem.
On Simulator the parser works well, but on a device it crashes all the time in my tests.
I feed the parser with this multipart-File https://dl.dropbox.com/u/12083655/delta-product-images-07-06-2012.bsr
and the crash (EXC_BAD_ACCESS code=1) occurs on the codeline MultipartParser:138 (c = buffer[i]).

Any Idea what is the problem here?
Can you reproduce it?

**Comment by johnyanarella** [comment]
Just saw this tonight - are you still seeing this issue (or were you able to resolve it since you reported it)?

**Comment by flandy84** [comment]
Yes, it's still an issue. This "array-out-of-bounds"-crash depends on the buffer-size of the data-chunks i feed the parser with. (depending on current network-connection type on the device the network-packages delivered varies in size. i download the file per http-request). with the combination of chunk-buffer-size of 16384 bytes and my posted Multipart-File, the parser crashes reproducible already on first data-chunk. with this check (c = (i<len)?buffer[i]:0;) i worked-around this crash, but don't know if this modification leads to other side-effects. Also don't know if there are other out-of-bounds issues with other file/chunk combinations in other parser-code-fragments
